### PR TITLE
check line-width of only newly added lines

### PR DESCRIPTION
**Pull request by gabreal**

### CODE REVIEW

File: scripts/gitlab/check_line_width.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+BASE_BRANCH="origin/master"
+LINE_WIDTH="101"
+
+
+if git diff ${BASE_BRANCH}...${CI_COMMIT_SHA} \
+  | grep "^.\{${LINE_WIDTH}\}" ${file}
+then
+  echo "|"
+  echo "| newly added lines longer than ${LINE_WIDTH} characters."
```
Comment (pepyakin) [inline scripts/gitlab/check_line_width.sh]: It is actually longer than 100 characters : P but I doooon't really care

File: scripts/gitlab/check_line_width.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+#
+# check if line width of rust source files is not beyond x characters
+#
+
+
+BASE_BRANCH="origin/master"
+LINE_WIDTH="101"
```
Comment (bkchr) [inline scripts/gitlab/check_line_width.sh]: We should accept the maximum of 120, 100 characters is more the preferred number.

Comment (pepyakin) [reply scripts/gitlab/check_line_width.sh]: Actually, the point of this check is exactly to notify people about exceeding the lower bound. 

Note, however, that this check doesn't block the merge it just acts as mere reminder to check if its possible to shorten the lines. 

Also, I'd say that if we set the limit to 120 here, we would effectively codify the 120 limit : )

Comment (bkchr) [reply scripts/gitlab/check_line_width.sh]: The problem is, that people are now already confused with the runtime check. They think that they need to make this happy and that they broke something. I think the same will happen with this one.

Comment (gabreal) [reply scripts/gitlab/check_line_width.sh]: maybe the text for these checks can be improved?

Comment (pepyakin) [reply scripts/gitlab/check_line_width.sh]: Ah, yeah, @bkchr you are right. I believe the runtime check and this check and potentially future checks that don't fail the build are actually warnings/lints and should be displayed accordingly.
Unfortunately, GitHub doesn't support it.

However, I think we already are treating them as warnings, i.e. not all changes to the runtime bumps the version (e.g. adding a comment under the srml directory).

But then, if people do get confused, we can, as @gabreal already noted, update the message to explicitly mention that:

1. this is a warning.
1. the rule which was violated, i.e. maximum 100 and 120 as hard limit,
1. the link to the style guide,

And the best thing, it cannot harm (unlike to the version check) if people try to make their lines below 100 not 120, if they after all above still perceive this as hard error : )

Comment (bkchr) [reply scripts/gitlab/check_line_width.sh]: Yeah sounds reasonable @pepyakin.

Comment (gavofyork) [reply scripts/gitlab/check_line_width.sh]: I like green ticks. There is nothing wrong with the occasional line being over 100 (and under 120). It is certainly not deserving of a red cross.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -82,7 +94,7 @@ test-linux-stable:                 &test
 check-web-wasm:
   stage:                           test
   image:                           tomaka/cargo-web:latest
-  allow_failure: true
+  allow_failure:                   true
```
Comment (gavofyork) [inline .gitlab-ci.yml]:
```suggestion
  allow_failure:                   false
```

i think it's fine to strictly require <= 120 characters now

File: scripts/gitlab/check_line_width.sh (hunk 1 above)
Comment (DemiMarie-temp) [reply scripts/gitlab/check_line_width.sh]:
```suggestion
  echo "| newly added lines longer than $((LINE_WIDTH - 1)) characters."
```

**Pull request merged**